### PR TITLE
Example app

### PR DESCRIPTION
**Pull request by ZeeshanGhalib**

### CODE REVIEW

File: components/command/command-handler/pom.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    <artifactId>command-handler</artifactId>
+
+    <dependencies>
+        <dependency>
+            <groupId>uk.gov.justice.services</groupId>
+            <artifactId>core</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
```
Comment (matt-rich) [inline components/command/command-handler/pom.xml]: We should replace all of these individual dependencies with the event-sourcing dependency that sits above this. May be suitable for a new task.

File: example/example-command/example-command-api/pom.xml [hunk 2]
```diff
@@ -14,13 +14,62 @@
...
     <build>
         <finalName>${project.artifactId}</finalName>
+        <plugins>
+            <plugin>
+                <groupId>uk.gov.justice.maven</groupId>
+                <artifactId>raml-maven-plugin</artifactId>
+                <version>${raml-maven-plugin.version}</version>
+                <configuration>
+                    <generatorName>uk.gov.justice.services.adapters.rest.generator.RestAdapterGenerator</generatorName>
+                    <sourceDirectory>${basedir}/src/raml</sourceDirectory>
```
Comment (matt-rich) [inline example/example-command/example-command-api/pom.xml]: Is it worth making these two values properties? - raml directories.

File: example/example-command/example-command-api/src/main/java/uk/gov/justice/services/example/cakeshop/command/api/MakeCakeCommandApi.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+public class MakeCakeCommandApi {
+
+    private static final Logger LOGGER = getLogger(MakeCakeCommandApi.class);
+
+    @Inject
+    Sender sender;
+
+    @Handles("cakeshop.commands.make-cake")
+    public void handle(final Envelope envelope) {
+        LOGGER.info("=============> Inside make-cake Command API");
```
Comment (matt-rich) [inline example/example-command/example-command-api/src/main/java/uk/gov/justice/services/example/cakeshop/command/api/MakeCakeCommandApi.java]: Don't you want to extract something from the payload to display here?

File: example/example-command/example-command-api/src/test/resources/arquillian.xml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<arquillian xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://jboss.org/schema/arquillian"
```
Comment (matt-rich) [inline example/example-command/example-command-api/src/test/resources/arquillian.xml]: Are you using Arquillian ?

File: example/example-command/example-command-api/src/test/resources/wildfly-config/standalone-test.xml [hunk 5]
```diff
@@ -0,0 +1,481 @@
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (matt-rich) [inline example/example-command/example-command-api/src/test/resources/wildfly-config/standalone-test.xml]: Is this part of Arquillian configuration above?

File: example/example-command/example-command-controller/pom.xml [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    <build>
+        <finalName>${project.artifactId}</finalName>
+        <plugins>
+            <plugin>
+                <groupId>uk.gov.justice.maven</groupId>
+                <artifactId>raml-maven-plugin</artifactId>
+                <version>${raml-maven-plugin.version}</version>
+                <configuration>
+                    <generatorName>uk.gov.justice.raml.jms.core.JmsEndpointGenerator</generatorName>
+                    <sourceDirectory>${basedir}/src/raml</sourceDirectory>
```
Comment (matt-rich) [inline example/example-command/example-command-controller/pom.xml]: As above, can this use the property values?

File: example/example-command/example-command-controller/src/raml/json/add-recipe.json [hunk 7]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (matt-rich) [inline example/example-command/example-command-controller/src/raml/json/add-recipe.json]: Do we need these files in multiple locations?

File: example/example-command/example-command-handler/pom.xml [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        <finalName>${project.artifactId}</finalName>
+
+        <plugins>
+            <plugin>
+                <groupId>uk.gov.justice.maven</groupId>
+                <artifactId>raml-maven-plugin</artifactId>
+                <version>${raml-maven-plugin.version}</version>
+                <configuration>
+                    <generatorName>uk.gov.justice.raml.jms.core.JmsEndpointGenerator</generatorName>
+                    <sourceDirectory>${basedir}/src/raml</sourceDirectory>
```
Comment (matt-rich) [inline example/example-command/example-command-handler/pom.xml]: properties as above

File: example/example-command/example-command-handler/src/main/java/uk/gov/justice/services/example/cakeshop/command/handler/AddRecipeCommandHandler.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    EventSource eventSource;
+
+    @Inject
+    TemporaryEventUtil temporaryEventUtil;
+
+    @Handles("cakeshop.commands.add-recipe")
+    public void addRecipe(final Envelope command) throws EventStreamException {
+        LOGGER.info("=============> Inside add-recipe Command Handler. RecipeId: " + command.payload().getString(FIELD_RECIPE_ID));
+
+        UUID recipeId = UUID.fromString(command.payload().getString(FIELD_RECIPE_ID));
```
Comment (matt-rich) [inline example/example-command/example-command-handler/src/main/java/uk/gov/justice/services/example/cakeshop/command/handler/AddRecipeCommandHandler.java]: final UUID

Comment (matt-rich) [inline example/example-command/example-command-handler/src/main/java/uk/gov/justice/services/example/cakeshop/command/handler/AddRecipeCommandHandler.java]: final UUID

File: example/example-command/example-command-handler/src/main/java/uk/gov/justice/services/example/cakeshop/command/handler/MakeCakeCommandHandler.java [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+import static uk.gov.justice.services.core.annotation.Component.COMMAND_HANDLER;
+
+@ServiceComponent(COMMAND_HANDLER)
+public class MakeCakeCommandHandler {
+
+    private static final Logger LOGGER = getLogger(MakeCakeCommandHandler.class);
+
+    @Handles("cakeshop.commands.make-cake")
+    public void handle(final Envelope envelope) throws EventStreamException {
+        LOGGER.info("=============> Inside make-cake Command Handler");
```
Comment (matt-rich) [inline example/example-command/example-command-handler/src/main/java/uk/gov/justice/services/example/cakeshop/command/handler/MakeCakeCommandHandler.java]: Add anything from the envelope for the log statement?

File: example/example-command/example-command-handler/src/raml/json/add-recipe.json [hunk 11]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (matt-rich) [inline example/example-command/example-command-handler/src/raml/json/add-recipe.json]: Again, does this need to be repeated?

File: example/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/AddRecipeCommandHandlerTest.java [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+
+    @Test
+    public void shouldHandleMakeCakeCommand() throws Exception {
+        final Envelope command = createCommand();
+        when(eventSource.getStreamById(RECIPE_ID)).thenReturn(eventStream);
+        when(temporaryEventUtil.eventsFrom(command)).thenReturn(events);
+
+        addRecipeCommandHandler.addRecipe(command);
+
+        verify(eventStream, times(1)).append(temporaryEventUtil.eventsFrom(command));
```
Comment (matt-rich) [inline example/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/AddRecipeCommandHandlerTest.java]: this should be verifying that the events is passed in, no?

File: example/example-event/example-event-listener/pom.xml [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+        <finalName>${project.artifactId}</finalName>
+
+        <plugins>
+            <plugin>
+                <groupId>uk.gov.justice.maven</groupId>
+                <artifactId>raml-maven-plugin</artifactId>
+                <version>${raml-maven-plugin.version}</version>
+                <configuration>
+                    <generatorName>uk.gov.justice.raml.jms.core.JmsEndpointGenerator</generatorName>
+                    <sourceDirectory>${basedir}/src/raml</sourceDirectory>
```
Comment (matt-rich) [inline example/example-event/example-event-listener/pom.xml]: Use properties as mentioned above.

File: example/example-event/example-event-listener/src/main/java/uk/gov/justice/services/example/cakeshop/event/listener/CakeMadeEventListener.java [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+
+@ServiceComponent(value = Component.EVENT_LISTENER)
+public class CakeMadeEventListener {
+
+    Logger logger = LoggerFactory.getLogger(CakeMadeEventListener.class);
+
+    @Handles("cakeshop.events.cake-made")
+    public void handle(final Envelope envelope) {
+
+        logger.info("=============> Inside cake-made Event Listener");
```
Comment (matt-rich) [inline example/example-event/example-event-listener/src/main/java/uk/gov/justice/services/example/cakeshop/event/listener/CakeMadeEventListener.java]: Add something from the envelope to log statement?

File: example/example-event/example-event-listener/src/raml/json/add-recipe.json [hunk 15]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (matt-rich) [inline example/example-event/example-event-listener/src/raml/json/add-recipe.json]: repeated again?

File: components/command/command-handler/pom.xml (hunk 1 above)
Comment (ZeeshanGhalib) [inline components/command/command-handler/pom.xml]: Yes, we will do this as a separate story/task.

File: example/example-command/example-command-api/src/main/java/uk/gov/justice/services/example/cakeshop/command/api/MakeCakeCommandApi.java (hunk 3 above)
Comment (ZeeshanGhalib) [inline example/example-command/example-command-api/src/main/java/uk/gov/justice/services/example/cakeshop/command/api/MakeCakeCommandApi.java]: We haven't really implemented make-cake yet.  Ony add-recipe.  Handlers have been created just to ensure it doesn't crash.

File: example/example-command/example-command-controller/pom.xml (hunk 6 above)
Comment (ZeeshanGhalib) [inline example/example-command/example-command-controller/pom.xml]: This section will be moved to the parent pom.

File: example/example-command/example-command-handler/pom.xml (hunk 8 above)
Comment (ZeeshanGhalib) [inline example/example-command/example-command-handler/pom.xml]: This section will be moved to the parent pom.

File: example/example-command/example-command-handler/src/main/java/uk/gov/justice/services/example/cakeshop/command/handler/MakeCakeCommandHandler.java (hunk 10 above)
Comment (ZeeshanGhalib) [inline example/example-command/example-command-handler/src/main/java/uk/gov/justice/services/example/cakeshop/command/handler/MakeCakeCommandHandler.java]: We haven't really implemented make-cake yet. Only add-recipe. Handlers have been created just to ensure it doesn't crash.

File: example/example-command/example-command-handler/src/raml/json/add-recipe.json (hunk 11 above)
Comment (ZeeshanGhalib) [inline example/example-command/example-command-handler/src/raml/json/add-recipe.json]: Yes

File: example/example-event/example-event-listener/pom.xml (hunk 13 above)
Comment (ZeeshanGhalib) [inline example/example-event/example-event-listener/pom.xml]: This section will be moved to the parent pom.

File: example/example-event/example-event-listener/src/raml/json/add-recipe.json (hunk 15 above)
Comment (ZeeshanGhalib) [inline example/example-event/example-event-listener/src/raml/json/add-recipe.json]: Yes

File: example/example-event/example-event-listener/src/main/java/uk/gov/justice/services/example/cakeshop/event/listener/CakeMadeEventListener.java (hunk 14 above)
Comment (ZeeshanGhalib) [inline example/example-event/example-event-listener/src/main/java/uk/gov/justice/services/example/cakeshop/event/listener/CakeMadeEventListener.java]: We haven't really implemented make-cake yet. Only add-recipe. Handlers have been created just to ensure it doesn't crash.

File: example/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/AddRecipeCommandHandlerTest.java [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+    @Mock
+    EventStream eventStream;
+
+    @Mock
+    TemporaryEventUtil temporaryEventUtil;
+
+    @Mock
+    Stream<Envelope> events;
+
+    @Mock
```
Comment (matt-rich) [inline example/example-command/example-command-handler/src/test/java/uk/gov/justice/services/example/cakeshop/command/handler/AddRecipeCommandHandlerTest.java]: I don't think we need the sender here anymore in the handler tier.

**Pull request merged**